### DISCUSSION TITLE
Why load event not working in v3

### DISCUSSION THREAD

**Discussion by ShekhSaifuddin007** [nuxt/nuxt#19198] [Questions]
In v3 I can't use this `load` event

```html
<script setup>
window.addEventListener('load', () => {
  alert('hello')
})     
</script>
```

**Comment by ndragun92** [comment]
If you use SSR you can do something like

```javascript
<script setup>
if (!import.meta.env.SSR) {
  window.addEventListener('load', () => {
    alert('hello');
  });
}
</script>

```

**Reply by ndragun92** [reply]
or `if (process.client)`

You most likely get an error because you tried to execute code on the server side and the server do not have access to the window context

Made an example for you:
https://stackblitz.com/edit/github-7skmnd?file=app.vue

**Reply by ShekhSaifuddin007** [reply]
I tried this in if (process.client) but mine did not work. In my `nuxt.config.ts` ssr is true

**Reply by ndragun92** [reply]
Not sure what else is in your project but you can see on provided stackblitz that it works

**Reply by ShekhSaifuddin007** [reply]
I tested it working fine without any layout when creating a layout then it not work anymore


https://stackblitz.com/edit/github-7skmnd-sgsrwm?file=nuxt.config.ts,app.vue,components%2FHello.vue,layouts%2Fdefault.vue

**Comment by TheAlexLichter** [comment]
If you use SSR, then the event listener registerd by nuxt will likely come in too late and the `load` event was already fired before Nuxt could finish the hydration. Instead, you could use the `onNuxtReady` composable.

**Reply by ShekhSaifuddin007** [reply]
Can you give me any example, actually this happens when I using nuxt layouts.

**Comment by SYPDung** [comment]
I'm having the same problem, load event doesn't work if I have nuxt layouts and nuxt pages. 

https://stackblitz.com/edit/nuxt-starter-4mvxas?file=app.vue
